packages/http/src/mocker/negotiator/NegotiatorHelpers.ts

```diff
@@ -39,8 +39,19 @@ const helpers = {
           mediaType,
           bodyExample: example,
         });
+      } else if (exampleKey === 'default' && httpContent.examples) {
+        return right({
```

Comment:
In an OAS3 context somebody might define a bunch of examples and the last one is default. This code would pick the first example instead of default, that’s not what we want. 

The feature is to support folks picking default regardless of OAS version. Http spec should be abstracting the differences, so that it doesn’t matter which version is coming in. 

We discussed this as one of two options to try but it looks like it won’t work. Instead of looking for “the first”, we probably want to look for one named default. If http spec needs to start labelling OAS2 examples with default then that’s an ok change with me.

Reply:
> In an OAS3 context somebody might define a bunch of examples and the last one is default. This code would pick the first example instead of default, that’s not what we want.

This is incorrect. This code will trigger only if there's not an example named `default`; it's a fallback.

The point is:

1. We have a test that's checking out that if there's no default example and I ask for `default` we should return 404
2. Another test that's checking that if there's no default example and I ask for a `default` we should return the first example.

Reply:
Here's an example of what I mean:

```
openapi: 3.0.2
paths:
  /todos:
    get:
      responses:
        200:
          description: Get Todo Items
          "content":
            application/json:
              example:
                id: 1
                name: turtle
              examples:
                cat:
                  summary: An example of a cat
                  value:
                    id: 2
                    name: cat
```

This should be identical to:

```
swagger: "2.0"
paths:
  /todos:
    get:
      produces:
        - application/json
      responses:
        200:
          description: Get Todo Items
          examples:
            application/json: 
              id: 1
              name: turtle
            cat: # this is the awkward hack discussed elsewhere
              id: 2
              name: cat
```

Ignoring the hack which we've employed in older tooling to give OAS2 named examples despite the spec not actually supporting it, the example named `application/json` should be treated the same as an in OAS3 under application/json and using the `example` keyword.

Does that make sense?

Reply:
> Another test that's checking that if there's no default example and I ask for a default we should return the first example.

Where is this requirement coming from? And what is the use case where asking for default should return first?

Reply:
I do not know :trollface: — I found it there and it's not passing, so I was wondering why's that happening


```
====test====
	Requesting a default example results in sending back the first and only example for an operation.
	====spec====
	swagger: "2.0"
	paths:
	  /todos:
	    get:
	      produces:
	        - application/json
	      responses:
	        200:
	          description: Get Todo Items
	          examples:
	            application/json: "hello"
	====server====
	mock -p 4010
	====command====
	curl -i -X GET http://localhost:4010/todos?__example=default
	====expect====
	HTTP/1.1 200 OK
	content-type: application/json; charset=utf-8
	
	"hello"
```

Reply:
There are so many things here 😢 

1. The OAS3 document you posted is actually invalid.

> The example object is mutually exclusive of the examples object.

https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#mediaTypeObject

Assuming this is not the issue

```yaml
swagger: "2.0"
paths:
  /todos:
    get:
      produces:
        - application/json
        - application/xml
      responses:
        200:
          description: Get Todo Items
          examples:
            application/json: 
              id: 1
              name: turtle
            application/xml: 
              id: 1
              name: turtle
            cat: # this is the awkward hack discussed elsewhere
              id: 2
              name: cat
```

@philsturgeon What would be the returned example, in this case? This would be equivalent, in the OAS3 example, to have *two* `example` — which we treat as the "default" one

Reply:
Ok yep its not valid OAS3 syntax I'm trying to highlight a bigger point. Let's do it without the named bits:

```
openapi: 3.0.2
paths:
  /todos:
    get:
      responses:
        200:
          description: Get Todo Items
          content:
            application/json:
              example:
                id: 1
                name: turtle
```

This should be identical to:

```
swagger: "2.0"
paths:
  /todos:
    get:
      produces:
        - application/json
      responses:
        200:
          description: Get Todo Items
          examples:
            application/json: 
              id: 1
              name: turtle
```

These should both be named "default", so ?__example=default should work the same in either.

The whole "just look for the first one" might get to the same result in this instance but it feels a bit like a hack to me, no?

what I'm getting at is the following should definitely not let ?__example=default return cat, it should be _the_ example for the mime type in question.

```
      responses:
        200:
          description: Get Todo Items
          examples:
            cat: 
              id: 2
              name: cat
            application/json: 
              id: 1
              name: turtle
```

OAS2 confuses names and mime types, and we kinda do with our hack to allow named examples, but we should not mess up default over that right?

Reply:
Or we just delete `?__example=default`. This whole conversation is making me wonder what the point of this feature even is?

Reply:
I struggle to see a scenario where I would explicitly ask for a `default` example — the default (the way I see it) is a fallback when I can't find anything suitable.

We can keep the functionality if you think it's useful; I will need to modify the http spec to "rename" the examples matching the `produces` into `defaults` and it should be good to go.

Your call!